pgmpy/models/LinearGaussianBayesianNetwork.py

```diff
@@ -272,18 +290,126 @@ def simulate(self, n_samples=1000, seed=None):
         >>> model.add_cpds(cpd1, cpd2, cpd3)
         >>> model.simulate(n_samples=500, seed=42)
         """
+        evidence = {} if evidence is None else evidence
+
+        do = {} if do is None else do
+
+        if set(do.keys()).intersection(set(evidence.keys())):
+            raise ValueError("Variable can't be in both do and evidence")
+
+        nodes = list(do.keys())
+
+        if not set(nodes).issubset(set(self.nodes())):
+            raise ValueError(
+                f"Nodes not found in the model: {set(nodes) - set(self.nodes)}"
+            )
+
         if len(self.cpds) != len(self.nodes()):
             raise ValueError(
                 "Each node in the model should have a CPD associated with it"
             )
 
-        mean, cov = self.to_joint_gaussian()
-        variables = list(nx.topological_sort(self))
+        if do != {}:
+            model = LinearGaussianBayesianNetwork(self.edges())
+            model.add_nodes_from(self.nodes())
+            model.add_cpds(*self.cpds)
+            for var, val in do.items():
```

Comment:
I don't think the logic here is correct. For a do operation in LinearGaussianBN, we will need to remove the do variables completely, and for each child variable of the do variable, set the beta parameter corresponding to the do variable to beta[i] * do-value.